packages/common/src/TransactionUtils.ts

```diff
@@ -19,6 +21,17 @@ export interface AugmentedSendOptions {
   maxPriorityFeePerGas?: number | string;
 }
 
+interface AugmentedWeb3 extends Web3 {
+  nonces: { [address: string]: number };
+}
+
+export interface executedTransaction {
+  receipt: TransactionReceipt;
```

Comment:
Why not make this an `|` type with a PromiEvent so we don't have to cast to `any`? Otherwise, callers will be expecting one type but receiving another.

Reply:
I tried using this syantax but this made it harder downstream in places that consume the output of the `runTransaction` function as they need to infer that status of the transaction based on the type. Rather, I'm going to add an additional prop to this interface: one for the `TransactionReceipt` and one for the `PromiEvent<TransactionReceipt>` so that we dont need to overload this param.

Reply:
on second thought your implementation is cleaner. the case work was pretty minimal won stream and could be fixed with a few simple casts.

Reply:
Other options:
1. [Overloading the method](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads). This would mean that if the compiler can deduce that the input value is true (i.e. it's hardcoded at the callsite), then it can deduce the correct type of the return value.
2. Creating two distinct methods, one that waits, one that doesn't.

Reply:
yup yup. I think the original proposal you had is the cleanest.